ports/mimxrt10xx/common-hal/microcontroller/Processor.c

```diff
@@ -57,6 +57,15 @@ uint32_t common_hal_mcu_processor_set_frequency(mcu_processor_obj_t *self,
     if (frequency < 24000000 || frequency > 1008000000) {
         mp_raise_ValueError(translate("Frequency Out of Range Must be between 24Mhz and 1.008Ghz"));
     }
+    if (frequency <= 600000000) {
+        mp_printf(&mp_plat_print, "System Clock is set to %d hz\n", frequency);
+    }
+    if (frequency > 600000000) {
+        mp_printf(&mp_plat_print, "System Clock is set to %d hz and is Overclocked\n", frequency);
+    }
+    if (frequency > 816000000) {
+        mp_printf(&mp_plat_print,  "System Clock is set to %d hz and is Overclocked, Cooling Required!\n", frequency);
```

Comment:
We don't generally print out warning messages, since it can generate "noise" on stdout or the REPL. If the user is doing something like this, we expect them to read the documentation and know what they are doing. So we'd prefer that you not include these logging-style messages.

If you want to protect against overclocking, then maybe we need to add a flag like `.overclocking_allowed`, or not use the property after all to set the frequency, but instead use a function call with an extra "it's ok to overclock" flag.

Reply:
Just thought it would be a good idea but since it doesn't match your typical usage I will revert the changes.  I did see mp_warning but did notice that you don't seem to use it much.  In testing you can overclock without an issue but if running at the highest speeds like 1Ghz you should probably add cooling but as you said you can add it in the documentation somewhere.